chore(ui-react-notifications): add deprecation message for InAppMessagingDisplay, InAppMessagingProvider and useInAppMessaging

sreeramsama opened the pull request:
<!--
Please make sure to read the Pull Request Guidelines:
https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md
-->

#### Description of changes
Add deprecation message for InAppMessagingDisplay, InAppMessagingProvider and useInAppMessaging
<!--
Thank you for your Pull Request! Please provide a description above and review
the requirements below.
-->

#### Description of how you validated changes

#### Checklist

<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

wlee221 reviewed (commented):
needs changeset. Maybe we can add a similar warning message and instructions in the changeset too?

sreeramsama commented:
Addressed comments by adding deprecation warnings to InAppMessagingDisplay, InAppMessagingProvider and useInAppMessaging, and adding changeset for the same.

wlee221 commented on .changeset/nasty-geckos-hide.md:
```diff
@@ -0,0 +1,7 @@
+"@aws-amplify/ui-react": patch
+---
+
+Add deprecation messages for InAppMessagingDisplay, InAppMessagingProvider and useInAppMessaging.
```
Can we add more details here? Some instructions on how customers can migrate would be great here.

sreeramsama replied:
Updated changeset to include more details.

wlee221 commented on .changeset/nasty-geckos-hide.md:
```diff
@@ -0,0 +1,7 @@
+"@aws-amplify/ui-react": patch
+---
+
+Add deprecation messages for InAppMessagingDisplay, InAppMessagingProvider and useInAppMessaging as they have been migrated to `@aws-amplify/ui-react-notifications` and will be removed from this package in a future major release. Please install `@aws-amplify/ui-react-notifications` and update the import paths.
```
nit:
```suggestion
Add deprecation messages for InAppMessagingDisplay, InAppMessagingProvider and useInAppMessaging. They have been migrated to `@aws-amplify/ui-react-notifications` and will be removed from this package in a future major release. Please install `@aws-amplify/ui-react-notifications` and update the import paths.
```

calebpollman commented on .changeset/nasty-geckos-hide.md:
```diff
@@ -0,0 +1,7 @@
+---
+"@aws-amplify/ui-react-core-notifications": patch
+"@aws-amplify/ui-react-core": patch
+"@aws-amplify/ui-react": patch
+---
```
Should we have a changeset all here? Shouldn't the primary PR against `main` contain one that covers everything?

wlee221 replied:
Oh yeah... that's fair. I'm good to move the changeset content to the primary one.

sreeramsama replied:
okay, will remove the one here, and add in the [main PR](https://github.com/aws-amplify/amplify-ui/pull/3824) with a message explaining all the changes.
